### PR TITLE
Fixup .gitignores to match repo reality

### PR DESCRIPTION
**Pull request by randall**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -43,7 +43,7 @@ Makefile-pl
 config.cache
 config.status
 config.nice
-config.h
+/config.h
```
Comment (maskit) [inline .gitignore:46]: Do we need this line? I don't have this file on my Mac and Ubuntu.

**Comment by randall** [comment]
[approve ci fedora]

File: .gitignore (hunk 1 above)
Comment (randall) [reply .gitignore]: you're right! We don't. Removed.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**